src/flownet/ahm/_assisted_history_matching.py

```diff
@@ -270,3 +277,448 @@ def save_iteration_parameters():
+    )
+    parser.add_argument("yamlobs", type=str, help="Path to the yaml observation file.")
+    parser.add_argument(
+        "start", type=str, help="Start date (YYYY-MM-DD) for accuracy analysis."
+    )
+    parser.add_argument(
+        "end", type=str, help="End date (YYYY-MM-DD) for accuracy analysis."
+    )
+    parser.add_argument(
+        "quantity",
+        type=str,
+        help="List of names of quantities of interest for accuracy analysis.",
+    )
+    parser.add_argument("metrics", type=str, help="List of names of accuracy metrics.")
+    parser.add_argument(
+        "outfile",
+        type=str,
+        help="Name of output file containing metrics over iterations.",
+    )
+    args = parser.parse_args()
+    args.runpath = args.runpath.replace("%d", "*")
+
+    print("Saving iteration analytics...", end=" ")
+
+    # Fix list inputs
+    keys = list(args.quantity.replace("[", "").replace("]", "").split(","))
+    metrics = list(args.metrics.replace("[", "").replace("]", "").split(","))
+
+    # Load ensemble of FlowNet
+    df, realizations_dict, key_list_data, iteration = make_dataframe_simulation_data(
+        args.runpath, args.eclbase, keys
+    )
+
+    # Load observation file (OPM reference / truth)
+    with open(args.yamlobs) as stream:
+        obs = {
+            item.pop("key"): item
+            for item in yaml.safe_load(stream).get("smry", [dict()])
+        }
+    df_obs = make_observation_dataframe(obs, key_list_data)
+
+    # Filter dataframe base on measurement dates
+    df = df[df["DATE"].isin(df_obs["DATE"])]
+
+    # Compute accuracy over iterations
+
+    # Initiate dataframe with metrics
+    df_metrics = load_csv_file(args.outfile, ["quantity", "iteration"] + metrics)
+
+    for str_key in keys:
+        truth_data = filter_dataframe(
+            df_obs, "DATE", np.datetime64(args.start), np.datetime64(args.end)
+        )
+        obs_opm = prepare_opm_reference_data(
+            truth_data, str_key, len(realizations_dict)
+        )
+
+        ens_flownet = []
+        ens_flownet.append(
+            prepare_flownet_data(
+                filter_dataframe(
+                    df, "DATE", np.datetime64(args.start), np.datetime64(args.end)
+                ),
+                str_key,
+                len(realizations_dict),
+            )
+        )
+
+        # Normalizing data
+        obs_opm, ens_flownet = normalize_data(obs_opm, ens_flownet)
+
+        # Appending dataframe with accuracy metrics of current iteration
+        dict_metric_tmp = compute_metric_ensemble(
+            obs_opm, ens_flownet, metrics, str_key, iteration
+        )
+        df_metrics = df_metrics.append(dict_metric_tmp, ignore_index=True)
+
+        # Plotting accuracy metrics over iterations
+        save_plots_metrics(df_metrics, metrics, str_key)
+
+    # Saving accuracy metrics to CSV file
+    df_metrics.to_csv(args.outfile + ".csv", index=False)
+
+    print("[Done]")
```

Comment:
The additions here does not appear to be tightly dependendt on existing code in `_assisted_history_matching`. Should we move it into a separate module (for readability)?

Reply:
I can put them in separate file, but they need to be in the same `ahm` module because this workflow is supposed to be run within ERT, thus in the same call of `run_flownet_history_matching` in `ahm._run_ahm.py`.

By the way, just added a commit to fix a typo in the function name (it was `run_flownet_history_mathing` previously).